### PR TITLE
Hoist class inheritance call so it runs before prototype use.

### PR DESCRIPTION
**Pull request by loganfsmyth**
As far as I can see, the constructor will always be a function declaration, so this is safe.

### CODE REVIEW

File: src/babel/transformation/transformers/es6/classes/vanilla.js [hunk 1]
```diff
@@ -592,11 +592,16 @@ export default class ClassTransformer {
...
-    this.pushedInherits = true;
-    this.body.push(t.expressionStatement(t.callExpression(
+    var inherits = t.expressionStatement(t.callExpression(
       this.file.addHelper("inherits"),
       [this.classRef, this.superName]
-    )));
+    ));
+    // Ensure that the constructor inheritance is set up before any properties
+    // can be assigned to the prototype.
+    inherits._blockHoist = 2;
```
Comment (sebmck) [inline src/babel/transformation/transformers/es6/classes/vanilla.js]: You should be able to just do `this.body.unshift` rather than using `_blockHoist`.

Comment (loganfsmyth) [inline src/babel/transformation/transformers/es6/classes/vanilla.js]: Oh right! Ha.

Comment (loganfsmyth) [inline src/babel/transformation/transformers/es6/classes/vanilla.js]: Fixed.

**Comment by sebmck** [comment]
Thanks @loganfsmyth!

**Pull request merged**